src/osd/PrimaryLogPG.cc

```diff
@@ -424,7 +424,7 @@ void PrimaryLogPG::on_local_recover(
       bool got = obc->get_recovery_read();
       assert(got);
 
-      assert(recovering.count(obc->obs.oi.soid));
+      assert(recovering.find(obc->obs.oi.soid) != recovering.end());
```

Comment:
This changes aren't helpful/useful.. the complexity is the same and they are more verbose.

The point is get rid of patterns like:

<pre>
if (foo.count(bar) && foo[bar] == baz) {
</pre>
where a count() precedes an operator[] call.  That is two lookups but can be reduced to 1 by using an iterator and comparing to end().